src/main/java/Storage.java

```diff
@@ -0,0 +1,129 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Handles all input and output related to the file associated with Duke.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Creates a new Storage which refers to the file of the given filepath.
+     * @param filepath used by Duke to store and modify information.
+     */
+    public Storage(String filepath) {
+        this.file = new File(filepath);
+    }
+
+    /**
+     * Saves the new list to the file.
+     * @param content of the modified list.
+     * @throws IOException Indicates that the given file to duke cannot be accessed.
+     */
+    public void saveToFile(String content) throws IOException {
+        FileWriter fw = new FileWriter(file);
+        fw.write(content);
+        fw.close();
+    }
+
+    /**
+     * Loads the saved tasks from the file to Duke's current task list.
+     * @return a list representing the tasks.
+     * @throws DukeException Indicating that the file given cannot be loaded.
+     */
+    public ArrayList<Task> loadPreviousTasks() throws DukeException {
+        ArrayList<Task> loadList = new ArrayList<>();
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            String str = "";
+            while ((str = br.readLine()) != null) {
+                Task task = txtProcessor(str);
+                loadList.add(task);
+            }
+        } catch (IOException e) {
+            throw new DukeException("File cannot be accessed");
+        }
+
+        return loadList;
+    }
+
+    /**
+     * Processes a line of text from the file referred to by Storage.
+     * @param line from the text file.
+     * @return the Task generated from this line.
+     */
+    private Task txtProcessor(String line) {
+        String details = line.substring(0,6);
+        char[] detailsArray = details.toCharArray();
+        String type = "" + detailsArray[1];
+        String desc = line.substring(7);
+
+        String tag = "";
+        boolean hasTags = false;
+        Task newTask = new Task("dummy");
+
+        // if there are tags, reprocess string and add them to task
+        if (desc.contains("tags")) {
+            hasTags = true;
+            desc = desc.replace("(", "");
+            desc = desc.replace(")", "");
+            String[] processed = desc.split("tags:");
+            desc = processed[0];
+            tag = processed[1];
+        }
+
+        if (type.equals("T")) {

```

Comment:
Is there a reason why your mutually exclusive condition checking is performed in separate if-blocks here? instead of using multiple if blocks here, you should change them to if-else or switch statements so that it's clear to a third party that these conditions (equals "T", equals "D" etc.) are mutually exclusive events and that only one branch should be taken